source/extensions/watchdog/abort_action/abort_action.cc

```diff
@@ -0,0 +1,69 @@
+#include "extensions/watchdog/abort_action/abort_action.h"
+
+#include "common/common/assert.h"
+#include "common/common/fmt.h"
+
+#ifndef WIN32
+#include <sys/types.h>
+#include <csignal>
+#endif
+
+#include "common/common/logger.h"
+#include "common/protobuf/utility.h"
+#include "envoy/thread/thread.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Watchdog {
+namespace AbortAction {
+namespace {
+#ifdef __linux__
+pid_t toPlatformTid(int64_t tid) { return static_cast<pid_t>(tid); }
+#elif defined(__APPLE__)
+uint64_t toPlatformTid(int64_t tid) { return static_cast<uint64_t>(tid); }
+#endif
+} // namespace
+
+AbortAction::AbortAction(
+    envoy::extensions::watchdog::abort_action::v3alpha::AbortActionConfig& config,
+    Server::Configuration::GuardDogActionFactoryContext& /*context*/)
+    : config_(config){};
+
+void AbortAction::run(
+    envoy::config::bootstrap::v3::Watchdog::WatchdogAction::WatchdogEvent /*event*/,
+    const std::vector<std::pair<Thread::ThreadId, MonotonicTime>>& thread_ltt_pairs,
```

Comment:
what is `ltt` stand for? Can you use something more descriptive please?

Reply:
Replaced it with `thread_last_checkin_pairs` hopefully that's a bit more descriptive; it's the last time the thread checked in with the watchdog system. I also changed where `ltt` was used to be `last_checkin` to be a more consistent.

Reply:
Yes, that's much clearer. Thanks!